File: JavaLibrary/src/main/java/com/flat/simplex/parser/logic/line/Line.java
package com.flat.simplex.parser.logic.line;

import com.flat.simplex.lexer.Token;
import com.flat.simplex.parser.logic.Block;

public class Line {

    protected Block parent;
    protected Token token;

    public Line(Block parent, Token token) {
        this.parent = parent;
        this.token = token;
    }

    public Token getToken() {
        return token;
    }

    public boolean isOp() {
        return false;
    }

    public LineOp getOp() {
        return null;
    }

    public LineValue getValue() {
        return null;
    }

    public Block getParent() {
        return parent;
    }
}

File: SimplexApp/simplex/parser/logic/line/Line.js
class Line {
    constructor(parent, token) {
        this.parent = parent;
        this.token = token;
    }

    getToken() {
        return this.token;
    }

    isOp() {
        return false;
    }

    getOp() {
        return null;
    }

    getValue() {
        return null;
    }

    getParent() {
        return this.parent;
    }
}

module.exports = Line;